From 1613dd4071c6d7291a88eed2a2938e2a94ee2849 Mon Sep 17 00:00:00 2001
From: glenvt18 <glenvt18@gmail.com>
Date: Fri, 4 Mar 2016 05:03:28 +0300
Subject: [PATCH] Add NEON support on AArch64

---
 configure.ac    | 7 ++++++-
 src/dvbcsa_pv.h | 5 +++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index c3c9c7d..3b45cd5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,6 +1,8 @@
 AC_INIT(libdvbcsa, 1.1.0)
 AC_PREREQ(2.50)
 
+AC_CANONICAL_HOST
+
 AC_ARG_ENABLE(debug, AC_HELP_STRING(--enable-debug, [Enable debug]), enable_debug=$enableval, enable_debug=no)
 
 if test "$enable_debug" = "yes" ; then
@@ -58,7 +60,10 @@ elif test "$enable_altivec" = "yes" ; then
 
 elif test "$enable_neon" = "yes" ; then
      AC_DEFINE(DVBCSA_USE_NEON, 1, Using NEON bitslice.)
-     GCC_CFLAGS="$GCC_CFLAGS -mfpu=neon"
+     case $host_cpu in
+         arm64*|aarch64*) ;;
+         arm*) GCC_CFLAGS="$GCC_CFLAGS -mfpu=neon" ;;
+     esac
 
 elif test "$enable_uint32" = "yes" ; then
      transpose_32=yes
diff --git a/src/dvbcsa_pv.h b/src/dvbcsa_pv.h
index d0ce8eb..2802179 100644
--- a/src/dvbcsa_pv.h
+++ b/src/dvbcsa_pv.h
@@ -95,6 +95,11 @@ void dvbcsa_key_schedule_block(const dvbcsa_cw_t cw, uint8_t * kk);
 #define DVBCSA_UNALIGNED_ACCESS_32 1
 #endif
 
+#if defined(__aarch64__)
+#define DVBCSA_UNALIGNED_ACCESS_32 1
+#define DVBCSA_UNALIGNED_ACCESS_64 1
+#endif
+
 DVBCSA_INLINE static inline void
 dvbcsa_xor_64 (uint8_t *b, const uint8_t *a)
 {
