--- a/CMakeLists.txt	2016-01-22 03:23:37.000000000 +0100
+++ b/CMakeLists.txt	2016-03-05 02:25:00.624648960 +0100
@@ -5,12 +5,6 @@
 set(LIBCEC_VERSION_MINOR 0)
 set(LIBCEC_VERSION_PATCH 1)
 
-find_program(CCACHE_FOUND ccache)
-if(CCACHE_FOUND)
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
-    set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
-endif(CCACHE_FOUND)
-
 # cec-client
 add_subdirectory(src/cec-client)
 add_dependencies(cec-client cec)
